src/main/java/org/jboss/pnc/konfluxtooling/logging/UploadLogCommand.java

```diff
@@ -0,0 +1,89 @@
+package org.jboss.pnc.konfluxtooling.logging;
+
+import java.io.File;
+import java.math.BigInteger;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.time.OffsetDateTime;
+
+import org.jboss.pnc.bifrost.upload.BifrostLogUploader;
+import org.jboss.pnc.bifrost.upload.LogMetadata;
+import org.jboss.pnc.bifrost.upload.TagOption;
+
+import io.quarkus.logging.Log;
+import picocli.CommandLine;
+
+@CommandLine.Command(name = "upload-log")
+public class UploadLogCommand implements Runnable {
+
+    private static final int DEFAULT_MAX_RETRIES = 4;
+
+    private static final int DEFAULT_DELAY_SECONDS = 60;
+
+    @CommandLine.Option(names = "--file", required = true)
+    String logFile;
+
+    @CommandLine.Option(names = "--bifrost-url", required = true)
+    String bifrostURL;
+
+    @CommandLine.Option(names = "--max-retries")
+    int maxRetries = DEFAULT_MAX_RETRIES;
+
+    @CommandLine.Option(names = "--delay-seconds")
+    int delaySeconds = DEFAULT_DELAY_SECONDS;
+
+    @CommandLine.Option(names = "--process-context", required = true)
+    String processContext;
+
+    @CommandLine.Option(names = "--process-context-variant", required = true)
+    String processContextVariant;
+
+    @CommandLine.Option(names = "--tmp", required = true)
+    String tmp;
+
+    @CommandLine.Option(names = "--request-context", required = true)
+    String requestContext;
+
+    public void run() {
+        try {
+            var logFilePath = Path.of(logFile);
+            var file = logFilePath.toFile();
+            if (!file.exists()) {
+                throw new RuntimeException(String.format(
+                        "No log file found at %s. Has the build been correctly done?", logFilePath));
+            }
+            var md5 = getMD5(logFilePath);
+            uploadLogsToBifrost(file, md5);
+        } catch (Exception e) {
+            Log.error("Upload log failed", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String getMD5(Path logFilePath) throws Exception {
+        byte[] data = Files.readAllBytes(logFilePath);
+        byte[] hash = MessageDigest.getInstance("MD5").digest(data);
+        return new BigInteger(1, hash).toString(16);
+    }
+
+    private void uploadLogsToBifrost(File logFile, String md5) {
+        BifrostLogUploader logUploader = new BifrostLogUploader(URI.create(bifrostURL),
+                maxRetries,
+                delaySeconds,
+                () -> System.getProperty("ACCESS_TOKEN"));
```

Comment:
If we're reusing DeployCommand then accessToken is a configProperty in https://github.com/project-ncl/konflux-tooling/blob/main/src/main/java/org/jboss/pnc/konfluxtooling/deploy/DeployCommand.java#L28

Reply:
How does the tool set value for "@ConfigProperty(name = "access.token")"? I did not see any Quarkus config file like application.properties or application.yaml. Or are there any way to provide the config file in runtime? And why do we need to use a config property? because I see the pipeline use an env var: `      env:
        - name: ACCESS_TOKEN
          value: $(params.ACCESS_TOKEN)`

Reply:
Quarkus detects it as an env var and converts it to a configuration property

Reply:
Thanks. I updated it as config property too.